Gordon360/Services/HousingService.cs

```diff
@@ -537,5 +560,215 @@ public bool ChangeApplicationDateSubmitted(int applicationID)
             _context.SaveChanges();
             return true;
         }
+
+        /// <summary>
+        /// update roommate imformation
+        /// </summary>
+        /// <param name="username"> The username for the user who complete the aplication form </param>
+        /// <param name="applicantion_id"> The ID of this application </param>
+        /// <param name="emailList"> A list of applicants' emails </param>
+        public async Task UpdateRoommateAsync(string username, string applicantion_id, string[] emailList)
+        {
+            var GordonID = _context.ACCOUNT.FirstOrDefault(a => a.AD_Username == username)?.gordon_id;
+            if (GordonID == null)
+            {
+                throw new ResourceNotFoundException() { ExceptionMessage = "The applicant could not be found" };
+            }
+            var existApplicant = _context.Applicant.Where(x => x.ApplicationID == applicantion_id);
+            var existMaxYear = _context.Year.Where(x => x.ApplicationID == applicantion_id);
+            foreach (Applicant a in existApplicant)
+            {
+                _context.Applicant.Remove(a);
+            }
+            foreach (Year y in existMaxYear)
+            {
+                _context.Year.Remove(y);
+            }
+            List<string> yearList = new List<string>();
+            foreach (string e in emailList)
+            {
+                if (e != "")
+                {
+                    var newApplicant = new Applicant
+                    {
+                        ApplicationID = applicantion_id,
+                        Applicant1 = e
+                    }; ;
+                    await _context.Applicant.AddAsync(newApplicant);
+
+                    var student = _context.Student.FirstOrDefault(x => x.Email == e);
+                    if (student != null)
+                    {
+                        yearList.Add(student.Class);
+                    }
+                }
+            }
+            var newMaxYear = new Year
+            {
+                ApplicationID = applicantion_id,
+                Year1 = yearList.Max()
+            }; ;
+            await _context.Year.AddAsync(newMaxYear);
+            _context.SaveChanges();
+        }
+
+        /// <summary>
+        /// update the information of preferred halls
+        /// </summary>
+        /// <param name="username"> The username for the user who complete the aplication form </param>
+        /// <param name="applicantion_id"> The ID of this application </param>
+        /// <param name="hallList"> A list of the preferred halls </param>
+        public async Task UpdateHallAsync(string username, string applicantion_id, string[] hallList)
```

Comment:
This name is a little bit misleading because `Hall` is not technically the object being modified here. It is `ApplicationHall`. A hall is its own object, so if I saw just the method name here, I would think this was to change something about, say, `Chase Hall` itself.